include/seastar/core/seastar.hh

```diff
@@ -359,6 +360,17 @@ using follow_symlink = bool_class<follow_symlink_tag>;
 /// with follow_symlink::yes, or for the link itself, with follow_symlink::no.
 future<stat_data> file_stat(std::string_view name, follow_symlink fs = follow_symlink::yes) noexcept;
 
+/// Returns details of the group from the group database.
+/// \param groupname name of the group
+///
+/// \return group_details of the group identified by name.
+future<group_details> grp_detail(std::string_view groupname) noexcept;
```

Comment:
Why noexcept? getgrnam() definitely can have errors - something as benign like a group that can't be found, to IO error reading /etc/group.

Reply:
The error can/should be returned in the future rather than having two ways to throw.